gitlab.go

```diff
@@ -69,6 +69,22 @@ const (
 	OwnerPermission      AccessLevelValue = 50
 )
 
+// ProtectedBranchAccessLevel represents access level for
+// protected branches within GitLab.
+//
+// GitLab API:
+// https://docs.gitlab.com/ce/api/protected_branches.html#protected-branches-api
+type ProtectedBranchAccessLevel int
+
+// List of available access levels for protected branches
+//
+// GitLab API: https://docs.gitlab.com/ce/api/protected_branches.html#protected-branches-api
+const (
+	NoAccess        ProtectedBranchAccessLevel = 0
+	DeveloperAccess ProtectedBranchAccessLevel = 30
+	MasterAccess    ProtectedBranchAccessLevel = 40
```

Comment:
These access levels are actually the same as the `AccessLevelValue` types. So I would only add the missing `NoPermissions` one and then reuse those instead of:

```
  NoPermissions     AccessLevelValue = 0
```

That would eliminate the use of these custom dedicated types...